common/crypto/platform/openssl/crypto.cpp

```diff
@@ -430,6 +439,97 @@ expected::ExpectedString SignRawData(
 	return Sign(private_key_path, shasum);
 }
 
+const size_t ecdsa256curveBits = 256;
+const size_t ecdsa256keySize = 32;
+
+// Try and decode the keys from pure binary, assuming that the points on the
+// curve (r,s), have been concatenated together (r || s), and simply dumped to
+// binary. Which is what we did in the `mender-artifact` tool.
+// (See MEN-1740) for some insight into previous issues, and the chosen fix.
+template <BIGNUM *DecoderFn(const unsigned char *, int, BIGNUM *)>
```

Comment:
Note that since this is a function pointer and not a type, you can save binary size by passing it as an argument instead.

Reply:
haha, shite. It's true, and I was so pleased with my template :roll_eyes:

Reply:
Indeed, the template use here ~is~ was very neat